src/app/(drawer)/(tabs)/substitutions.tsx

```diff
@@ -1,10 +1,112 @@
 import ScreenWrapper from "@/components/ScreenWrapper"
-import { Text } from "react-native"
+import Heading from "@/components/ui/Heading"
+import IconButton from "@/components/ui/IconButton"
+import { useSubstitutions } from "@/hooks/substitutions/useSubstitutions"
+import useColors from "@/hooks/useColors"
+import { useQueryClient } from "@tanstack/react-query"
+import { addDays, format, subDays } from "date-fns"
+import { pl } from "date-fns/locale/pl"
+import { ChevronLeft, ChevronRight } from "lucide-react-native"
+import { useCallback, useState } from "react"
+import { RefreshControl, ScrollView, Text, View } from "react-native"
 
 const Substitutions = () => {
+  const [date, setDate] = useState(new Date())
+  const [refreshing, setRefreshing] = useState(false)
+
+  const queryClient = useQueryClient()
+  const colors = useColors()
+
+  const { data, isLoading, isError, isRefetching, fetchNextPage, isFetching } =
+    useSubstitutions({ date })
+
+  const onRefresh = useCallback(() => {
+    setRefreshing(true)
+    queryClient
```

Comment:
const { refetch } = useSubstitutions({ date })
czy tego nie da się zrobić za pomoca refetch?

Reply:
Nie, bo wtedy refetuchuje sie tylko ten dzien na ktorym jestesmy a nie wszystkie

Reply:
a zoba to
w hooku można dodać funkcję w `onBeforeRefetch `
żeby nie robić tego w widoku
https://github.com/TanStack/query/discussions/5692#discussioncomment-6448904

```ts
const resetInfiniteQueryPagination = (): void => {
  queryClient.setQueryData(queryKey, (oldData) => {
    if (!oldData) return undefined;

    return {
      ...oldData,
      pages: oldData.pages.slice(0, 1),
      pageParams: oldData.pageParams.slice(0, 1),
    };
  });
};
```